pkg/build/rpm_test.go

```diff
@@ -117,8 +124,142 @@ func TestCopyRPMsNoRPMDir(t *testing.T) {
 	require.NoError(t, err)
 	defer os.Remove(builder.eibBuildDir)
 
+	rpmDestDir := builder.combustionDir
+	rpmSourceDir, err := builder.generateRPMPath()
+	require.NoError(t, err)
+
+	// Test
+	err = copyRPMs(rpmSourceDir, rpmDestDir, nil)
+
+	// Verify
+	require.NoError(t, err)
+}
+
+func TestWriteRPMScript(t *testing.T) {
+	// Setup
+	tmpDir, err := os.MkdirTemp("", "eib-")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	buildConfig := config.BuildConfig{
+		ImageConfigDir: "../config/testdata",
+	}
+	builder := New(nil, &buildConfig)
+	require.NoError(t, builder.prepareBuildDir())
+
+	rpmSourceDir, err := builder.generateRPMPath()
+	require.NoError(t, err)
+
+	file1Path := filepath.Join(rpmSourceDir, "rpm1.rpm")
+	file1, err := os.Create(file1Path)
+	require.NoError(t, err)
+	defer os.Remove(file1Path)
+
+	file2Path := filepath.Join(rpmSourceDir, "rpm2.rpm")
+	file2, err := os.Create(file2Path)
+	require.NoError(t, err)
+	defer os.Remove(file2Path)
+
+	rpmFileNames, err := getRPMFileNames(rpmSourceDir)
+	require.NoError(t, err)
+
+	// Test
+	err = builder.writeRPMScript(rpmFileNames)
+
+	// Verify
+	require.NoError(t, err)
+
+	expectedFilename := filepath.Join(builder.combustionDir, modifyRPMScriptName)
+	foundBytes, err := os.ReadFile(expectedFilename)
+	require.NoError(t, err)
+
+	stats, err := os.Stat(expectedFilename)
+	require.NoError(t, err)
+	assert.Equal(t, fs.FileMode(modifyScriptMode), stats.Mode())
+
+	foundContents := string(foundBytes)
+	assert.Contains(t, foundContents, "rpm1.rpm")
+	assert.Contains(t, foundContents, "rpm2.rpm")
+
+	// Cleanup
+	assert.NoError(t, file1.Close())
+	assert.NoError(t, file2.Close())
+}
+
+func TestProcessRPMs(t *testing.T) {
+	// Setup
+	bc := config.BuildConfig{
+		ImageConfigDir: "../config/testdata",
+	}
+	builder := New(nil, &bc)
+	err := builder.prepareBuildDir()
+	require.NoError(t, err)
+	defer os.Remove(builder.eibBuildDir)
+
+	rpmDestDir := builder.combustionDir
```

Comment:
Dropping this variable is actually a cleaner indicator for the assertions. If we'd prefer to keep it instead, then it should be moved to where it is actually used.

Reply:
Sounds good, will update.